src/main/java/book/store/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,107 @@
+package book.store.controller;
+
+import book.store.dto.cart.item.CartItemRequestDto;
+import book.store.dto.cart.item.CartItemUpdateQuantityDto;
+import book.store.dto.shopping.cart.ShoppingCartResponseDto;
+import book.store.service.shopping.cart.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Shopping Cart Controller Management",
+        description = "Endpoints for managing items in Shopping Carts")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @Operation(summary = "Add new item to a shopping cart",
+            description = "Adds a new item to the user's shopping cart, "
+                    + "with quantity specified in the request.")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200",
+                    description = "Item successfully added to cart"),
+            @ApiResponse(responseCode = "400",
+                    description = "Invalid item or quantity specified"),
+            @ApiResponse(responseCode = "401",
+                    description = "Unauthorized. Authentication required"),
+            @ApiResponse(responseCode = "403",
+                    description = "Forbidden. Permission denied")
+    })
+    @PreAuthorize("hasRole('USER')")
+    @PostMapping
+    public ShoppingCartResponseDto addCartItem(
```

Comment:
add to every endpoint
```suggestion
    public ShoppingCartResponseDto addCartItem(Authentication authentication,
```